src/components/ResetPassword.jsx

```diff
@@ -1,13 +1,59 @@
-import { useState } from "react"
-import { Link, useParams } from "react-router-dom"
+import { useEffect, useState } from "react"
+import {useDispatch, useSelector} from 'react-redux'
+import { Link, useParams, useNavigate } from "react-router-dom"
+import {resetPassword, reset} from '../redux/userSlice'
+import { toast } from "react-toastify";
 
-const ResetPassword = () => {
 
+const ResetPassword = () => {
+  const {id, token} = useParams();
   const [password, setPassword] = useState('');
   const [confirmPassword, setConfirmPassword] = useState('');
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  
+  const {isError, isSuccess, message} = useSelector(state => state.user)
+
+  useEffect(() => {
+    let subscribed = true;
+
+    if(isSuccess){
+      toast.success(message || "Password Updated");
+      navigate('/user/login')
+    }
 
+    else if(isError){
+      toast.error(message || "Something went wrong")
+    }
+
+    return () => {
+      dispatch(reset())
+      subscribed = false;
+    }
+  },[isError, isSuccess])
+
+  //handle submit button action
   function handleSubmit(e){
+    e.preventDefault()
+    const passwordRegex = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/
```

Comment:
use password regex from constant files, So it makes reusable in other components also if needed.